cpp/src/arrow/type.h

```diff
@@ -1657,6 +1657,11 @@ class ARROW_EXPORT FieldRef {
   const std::string* name() const {
     return IsName() ? &util::get<std::string>(impl_) : NULLPTR;
   }
+  const std::vector<FieldRef>* nested_refs() const {
+    return util::holds_alternative<std::vector<FieldRef>>(impl_)
+               ? &util::get<std::vector<FieldRef>>(impl_)
+               : NULLPTR;
+  }
```

Comment:
Ah, this is because IsNested is checking whether it's either a FieldPath or a series of Names, but this accessor only wants the latter case. (I think the IsNested naming is a little unfortunate…)